src/components/AttachmentPicker/index.native.js

```diff
@@ -250,136 +148,207 @@ class AttachmentPicker extends Component {
-    }
+        })
+
+    const [menuItemData, setMenuItemData] = useState([
+        {
+            icon: Expensicons.Camera,
+            textTranslationKey: 'attachmentPicker.takePhoto',
+            pickAttachment: () => showImagePicker(launchCamera),
+        },
+        {
+            icon: Expensicons.Gallery,
+            textTranslationKey: 'attachmentPicker.chooseFromGallery',
+            pickAttachment: () => showImagePicker(launchImageLibrary),
+        },
+    ]);
 
     /**
      * An attachment error dialog when user selected malformed images
      */
-    showImageCorruptionAlert() {
-        Alert.alert(this.props.translate('attachmentPicker.attachmentError'), this.props.translate('attachmentPicker.errorWhileSelectingCorruptedImage'));
-    }
+    const showImageCorruptionAlert = useCallback(() => {
+        Alert.alert(translate('attachmentPicker.attachmentError'), translate('attachmentPicker.errorWhileSelectingCorruptedImage'));
+    }, [translate])
 
     /**
      * Launch the DocumentPicker. Results are in the same format as ImagePicker
      *
      * @returns {Promise<DocumentPickerResponse[]>}
      */
-    showDocumentPicker() {
-        return RNDocumentPicker.pick(documentPickerOptions).catch((error) => {
+    const showDocumentPicker = () => RNDocumentPicker.pick(documentPickerOptions).catch((error) => {
             if (RNDocumentPicker.isCancel(error)) {
                 return;
             }
 
-            this.showGeneralAlert(error.message);
+            showGeneralAlert(error.message);
             throw error;
-        });
-    }
-
-    /**
-     * Triggers the `onPicked` callback with the selected attachment
-     */
-    completeAttachmentSelection() {
-        if (!this.state.result) {
-            return;
-        }
-
-        this.state.onPicked(this.state.result);
-    }
+        })
 
     /**
      * Opens the attachment modal
      *
-     * @param {function} onPicked A callback that will be called with the selected attachment
+     * @param {function} onPickedHandler A callback that will be called with the selected attachment
      */
-    open(onPicked) {
-        this.completeAttachmentSelection = onPicked;
-        this.setState({isVisible: true});
+    const open = (onPickedHandler) => {
+        completeAttachmentSelection.current = onPickedHandler;
+        setIsVisible(true);
     }
 
     /**
      * Closes the attachment modal
      */
-    close() {
-        this.setState({isVisible: false});
+    const close = () => {
+        setIsVisible(false);
     }
 
+    /**
+     * Handles the image/document picker result and
+     * sends the selected attachment to the caller (parent component)
+     *
+     * @param {Array<ImagePickerResponse|DocumentPickerResponse>} attachments
+     * @returns {Promise}
+     */
+    const pickAttachment = useCallback((attachments = []) => {
+        if (attachments.length === 0) {
+            return;
```

Comment:
Should this return a promise with some null value? At least that's what the function description says 😅

Reply:
This method is copied 1:1 from the class implementation, do you think we should change it? Or at least change the comment?

Reply:
You did a good copy, it's the logic itself which I don't think it's the best 😅

I would change it to always return a promise with a value, this way the function respects its contract and a developer knows that no matter what happens inside of it, it will always return some promise 😄